NSURL.m

```diff
@@ -156,6 +157,19 @@ static void posixError(CFErrorRef *error) {
     CFRelease(err);
 }
 
+static void dlError(CFErrorRef *error){
+    const CFStringRef keys[1] = {
+        kCFErrorUnderlyingErrorKey,
+    };
+    
+    CFStringRef err = CFStringCreateWithFormat(kCFAllocatorDefault, NULL, CFSTR("%s"), dlerror());
```

Comment:
Why go through the formatting machinery when you can just create it from the C string directly?

```suggestion
    CFStringRef underlyingError = CFStringCreateWithCString(kCFAllocatorDefault, dlerror(), kCFStringEncodingUTF8);
```